### PR TITLE
Fix bug when doing outer join on multi-dim columns

### PR DESCRIPTION
**Pull request by taldcroft**
This fixes a bug reported in #4059.

@embray - would like to get this in 1.0.4 if possible.

### CODE REVIEW

File: astropy/table/operations.py [hunk 1]
```diff
@@ -636,8 +636,14 @@ def _join(left, right, keys=None, join_type='inner',
         # If the output table is masked then set the output column masking
         # accordingly.  Check for columns that don't support a mask attribute.
         if masked:
+            # array_mask is 1-d corresponding to length of output column.  We need
+            # make it have the correct shape for broadcasting, ie. (length, 1, 1, ..).
+            # Mixin columns might not have ndim attribute so use len(col.shape).
+            mask_shape = [out[out_name].shape[0]] + [1] * (len(out[out_name].shape) - 1)
```
Comment (mhvk) [inline astropy/table/operations.py]: @taldcroft - here, one could just write
```
array_mask.shape = (out[out_name].shape[0],) + (1,) * (len(out[out_name].shape) - 1)
```
This is one line shorter and slightly faster and I think just as clear about intent.  Also, I always think of the shape as a tuple, so am using that; it also happens to be substantially faster than lists (though likely in the noise here):
```
In [2]: %timeit (10,) + (5-1)*(1,)
100000000 loops, best of 3: 19.6 ns per loop

In [3]: %timeit [10,] + (5-1)*[1]
1000000 loops, best of 3: 645 ns per loop
```

**Comment by mhvk** [comment]
@taldcroft - the fix looks good and I only put a minor comment, which you are free to ignore if you wish.  As I'm on holidays, I'm only sporadically checking e-mail, so feel free just to merge.

**Comment by mhvk** [comment]
@taldcroft - the fix looks good and I only put a minor comment, which you are free to ignore if you wish.  As I'm on holidays, I'm only sporadically checking e-mail, so feel free just to merge.

**Comment by embray** [comment]
This is going to go in v1.0.5.  I already finished the testing for v1.0.4 on Friday and plan to make the release today (I just didn't have time on Friday).

**Comment by embray** [comment]
Sorry 'bout that--I should have communicated the status of the release process better.  On the up side I'm going to try to get v1.0.5 out a lot faster after .4

**Comment by taldcroft** [comment]
No worries @embray.

**Comment by astrofrog** [comment]
@taldcroft @embray - note that this was never backported, but I just did it now.

**Comment by bsipocz** [comment]
@astrofrog - Doesn't it also mean that the milestone should be changed?

**Comment by astrofrog** [comment]
@bsipocz - good catch, thanks!

**Pull request merged**